test/services/database/post_repository_test.dart

```diff
@@ -233,5 +233,17 @@ void main() {
       expect(actualPosts1, unorderedEquals([postsData1, postsData2]));
       expect(actualPosts2, unorderedEquals([postsData3, postsData4]));
     });
+
+    test("post exists works", () async {
+      await setPostFirestore(post);
+
+      final exists = await postRepository.postExists(post.id);
+      expect(exists, true);
+
+      await postRepository.deletePost(post.id);
+
+      final existAfterDeletion = await postRepository.postExists(post.id);
+      expect(existAfterDeletion, false);
+    });
```

Comment:
I think verifying the post does not exist before its creation adds some useful robustness to the test (this allows to test a post does not exist when it has been deleted, but also when it has never existed). I therefore propose:
```suggestion
    test("post exists works", () async {
      final existsBeforeCreation = await postRepository.postExists(post.id);
      expect(existsBeforeCreation, false);

      await setPostFirestore(post);

      final exists = await postRepository.postExists(post.id);
      expect(exists, true);

      await postRepository.deletePost(post.id);

      final existAfterDeletion = await postRepository.postExists(post.id);
      expect(existAfterDeletion, false);
    });
```